### PR TITLE
System.Data.SQLite -> Microsoft.Data.Sqlite

### PR DESCRIPTION
**Pull request by CasualPokePlayer**
https://learn.microsoft.com/en-us/dotnet/standard/data/sqlite/compare

Seems to be a sort of successor / "modern variant" of System.Data.SQLite? netstandard2.0 so great for us I suppose. I assume it works for Linux? (System.Data.SQLite didn't lol). A quick testing of commands seems to say it works on Windows at least.

Also would eliminate another .NET Framework dep (see #2261).

Check if completed:
- [ ] I have run any relevant test suites
- [x] I, the committer, have read the [licensing terms for contributors](https://github.com/TASEmulators/BizHawk/blob/master/contributing.md#copyrights-and-licensing) (last updated 2022-07-15) and am compliant

### CODE REVIEW

File: src/BizHawk.Client.Common/Api/Classes/SQLiteApi.cs [hunk 1]
```diff
@@ -1,42 +1,45 @@
...
 	{
-		private SQLiteConnection _dbConnection;
+		private SqliteConnection _dbConnection;
 
 		public string CreateDatabase(string name)
 		{
 			try
 			{
-				SQLiteConnection.CreateFile(name);
+				File.Create(name).Dispose();
```
Comment (YoshiRulz) [inline src/BizHawk.Client.Common/Api/Classes/SQLiteApi.cs:17]: If the equivalent of this in the new library is a no-op, there's no point checking it. We assume paths are good and let them throw on first use.

**Review by YoshiRulz — APPROVED** [review]

**Comment by YoshiRulz** [comment]
Oh I probably shouldn't approve without testing on Linux, huh.

File: src/BizHawk.Client.Common/Api/Classes/SQLiteApi.cs (hunk 1 above)
Comment (CasualPokePlayer) [reply src/BizHawk.Client.Common/Api/Classes/SQLiteApi.cs:17]: This is the equivalent of what the old library did (well, old library did `File.Create(name).Close();` technically, but Dispose() is probably more proper)

**Comment by YoshiRulz** [comment]
Works under Mono on NixOS, without host SQLite lib even.

**Pull request merged**